nuget/updater/main.ps1

```diff
@@ -0,0 +1,86 @@
+Set-StrictMode -version 2.0
+$ErrorActionPreference = "Stop"
+
+$jobString = Get-Content -Path $env:DEPENDABOT_JOB_PATH
+$job = (ConvertFrom-Json -InputObject $jobString).job
+
+function Get-Files {
+    Write-Host "Job: $($job | ConvertTo-Json)"
+    $sourceRepo = $job.source.repo
+    # TODO: handle other values from $job.source.provider
+    $url = "https://github.com/$sourceRepo"
+    $branchArgs = ""
+    if ("branch" -in $job.source.Psobject.Properties.Name) {
+        $branchArgs = "--branch $($job.source.branch)"
+    }
+
+    Invoke-Expression "git clone --recurse-submodule $branchArgs $url $env:DEPENDABOT_REPO_CONTENTS_PATH"
+    if ("commit" -in $job.source.PSobject.Properties.Name) {
+        git fetch origin
+        git checkout $job.source.commit
+    }
```

Comment:
I think we should rework this to do very shallow clones.

If `$job.source.commit` is set then something like
```pwsh
$currentDirectory = Get-Location
New-Item -ItemType "directory" -Path $env:DEPENDABOT_REPO_CONTENTS_PATH
Set-Location $env:DEPENDABOT_REPO_CONTENTS_PATH
git init
git add remote origin $url
git fetch --depth=1 --no-tags origin $job.source.commit
git checkout FETCH_HEAD
git submodule update --init --recursive
Set-Location $currentDirectory
```
ElIf `job.source.branch` is set then something like
```pwsh
git clone --depth=1 --no-tags $url --single-branch --branch $job.source.branch --recurse-submodule
```
Else no branch specified then something like
```pwsh
git clone --depth=1 --no-tags $url --single-branch --recurse-submodule
```

Reply:
Good call.  The dependabot cli already does the `git init` and `git remote add` etc., but it's up to us to do the fetch.  I'll adopt your changes.